alibuild_helpers/build.py

```diff
@@ -1205,7 +1214,11 @@ def downloadTask(p):
 
     dieOnError(err, buildErrMsg)
 
-    syncHelper.syncToRemote(p, spec)
+    # Make sure not to upload local-only packages! These might have been
+    # produced in a previous run with a read-only remote store.
+    if not spec["revision"].startswith("local"):
+      syncHelper.syncToRemote(p, spec)
```

Comment:
Can we invert the logic here? Something like:

```
if spec["revision"].startswith("local"):
  continue       # break?
```

Reply:
Done.